packages/material-ui/src/ClickAwayListener/ClickAwayListener.test.js

```diff
@@ -185,7 +185,7 @@ describe('<ClickAwayListener />', () => {
         }
         render(<Test />);
 
-        fireDiscreteEvent.click(screen.getByTestId('trigger'));
+        fireEvent.click(screen.getByTestId('trigger'));
```

Comment:
Ran test suite without the fix in https://github.com/mui-org/material-ui/pull/23315 and the test still passes. When `act` was batching discrete updates the test passed which was not desired.